File: medplum-java/medplum-model/src/main/java/com/fhir/ExplanationOfBenefit_Procedure.java
package com.fhir;

import com.StagedBuilderStyle;
import com.fasterxml.jackson.annotation.JsonIgnoreProperties;
import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import java.lang.String;
import java.util.List;
import java.util.Optional;
import org.immutables.value.Value;

@Value.Immutable
@StagedBuilderStyle
@JsonDeserialize(
    as = ImmutableExplanationOfBenefit_Procedure.class
)
@JsonIgnoreProperties(
    ignoreUnknown = true
)
public interface ExplanationOfBenefit_Procedure {
  Optional<List<Extension>> extension();

  Optional<List<CodeableConcept>> type();

  Optional<dateTime> date();

  Optional<Reference> procedureReference();

  Optional<CodeableConcept> procedureCodeableConcept();

  Optional<positiveInt> sequence();

  Optional<List<Reference>> udi();

  Optional<String> id();

  Optional<List<Extension>> modifierExtension();

  static ImmutableExplanationOfBenefit_Procedure.Builder builder() {
    return ImmutableExplanationOfBenefit_Procedure.builder();
  }
}

File: medplum-typescript/packages/sdk/src/model/fhir/ExplanationOfBenefit_Procedure.ts
import { CodeableConcept } from "./CodeableConcept";
import { dateTime } from "./dateTime";
import { Extension } from "./Extension";
import { positiveInt } from "./positiveInt";
import { Reference } from "./Reference";

export interface ExplanationOfBenefit_Procedure {
    id: string | null | undefined;
    extension: Extension[] | null | undefined;
    modifierExtension: Extension[] | null | undefined;
    sequence: positiveInt | null | undefined;
    type: CodeableConcept[] | null | undefined;
    date: dateTime | null | undefined;
    procedureCodeableConcept: CodeableConcept | null | undefined;
    procedureReference: Reference | null | undefined;
    udi: Reference[] | null | undefined;
}
